src/Avalonia.Input/DragEventArgs.cs

```diff
@@ -1,17 +1,40 @@
-﻿using Avalonia.Interactivity;
+﻿using System;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
 
 namespace Avalonia.Input
 {
     public class DragEventArgs : RoutedEventArgs
     {
+        private Interactive _target;
+        private Point _targetLocation;
+
         public DragDropEffects DragEffects { get; set; }
 
         public IDataObject Data { get; private set; }
 
-        public DragEventArgs(RoutedEvent<DragEventArgs> routedEvent, IDataObject data)
+        public Point GetPosition(IVisual relativeTo)
+        {
+            var point = new Point(0, 0);
+
+            if (relativeTo == null)
+            {
+                throw new ArgumentNullException(nameof(relativeTo));
+            }
+
+            if (_target != null)
+            {
+                point = _target.TranslatePoint(_targetLocation, relativeTo);
```

Comment:
So if `_target` is null, then the point won't be translated to `relativeTo`? Could this cause surprising behavior? In what circumstances will `_target` be null?

Reply:
When https://github.com/wieslawsoltes/Avalonia/blob/ee0c3dfa94f750db1657d93047f704ee2a736dd1/src/Avalonia.Input/DragDropDevice.cs#L14 this returns null.

I was not sure what to return in this case.

Reply:
Maybe I just should return `_targetLocation`?

Reply:
I thinks its ok because the no drag event is called when `_target` is null.

Reply:
@grokys You can test this in DragAndDropPage :
```C#
using Avalonia.Controls;
using Avalonia.Input;
using Avalonia.Markup.Xaml;
using System;
using System.Collections.Generic;
using System.Text;

namespace ControlCatalog.Pages
{
    public class DragAndDropPage : UserControl
    {
        private TextBlock _DropState;
        private TextBlock _DragState;
        private Border _DragMe;
        private int DragCount = 0;

        public DragAndDropPage()
        {
            this.InitializeComponent();

            _DragMe.PointerPressed += DoDrag;

            AddHandler(DragDrop.DropEvent, Drop);
            AddHandler(DragDrop.DragEnterEvent, DragEnter);
            AddHandler(DragDrop.DragOverEvent, DragOver);
        }

        private async void DoDrag(object sender, Avalonia.Input.PointerPressedEventArgs e)
        {
            DataObject dragData = new DataObject();
            dragData.Set(DataFormats.Text, $"You have dragged text {++DragCount} times");

            var result = await DragDrop.DoDragDrop(dragData, DragDropEffects.Copy);
            switch(result)
            {
                case DragDropEffects.Copy:
                    _DragState.Text = "The text was copied"; break;
                case DragDropEffects.Link:
                    _DragState.Text = "The text was linked"; break;
                case DragDropEffects.None:
                    _DragState.Text = "The drag operation was canceled"; break;
            }
        }

        private void DragEnter(object sender, DragEventArgs e)
        {
            // Only allow Copy or Link as Drop Operations.
            e.DragEffects = e.DragEffects & (DragDropEffects.Copy | DragDropEffects.Link);

            Console.WriteLine($"DragEnter {e.GetPosition(e.Source as IControl)}");

            // Only allow if the dragged data contains text or filenames.
            if (!e.Data.Contains(DataFormats.Text) && !e.Data.Contains(DataFormats.FileNames))
                e.DragEffects = DragDropEffects.None;
        }

        private void DragOver(object sender, DragEventArgs e)
        {
            // Only allow Copy or Link as Drop Operations.
            e.DragEffects = e.DragEffects & (DragDropEffects.Copy | DragDropEffects.Link);

            Console.WriteLine($"DragOver {e.GetPosition(e.Source as IControl)}");

            // Only allow if the dragged data contains text or filenames.
            if (!e.Data.Contains(DataFormats.Text) && !e.Data.Contains(DataFormats.FileNames))
                e.DragEffects = DragDropEffects.None; 
        }

        private void Drop(object sender, DragEventArgs e)
        {
            Console.WriteLine($"Drop {e.GetPosition(e.Source as IControl)}");

            if (e.Data.Contains(DataFormats.Text))
                _DropState.Text = e.Data.GetText();
            else if (e.Data.Contains(DataFormats.FileNames))
                _DropState.Text = string.Join(Environment.NewLine, e.Data.GetFileNames());
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);

            _DropState = this.Find<TextBlock>("DropState");
            _DragState = this.Find<TextBlock>("DragState");
            _DragMe = this.Find<Border>("DragMe");
        }
    }
}
```